### PR TITLE
Package: Update dependencies and Travis build config

### PR DESCRIPTION
**Pull request by DevBonBon**
See commit messages for more details.

### CODE REVIEW

**Comment by DevBonBon** [comment]
I feel like the Travis build is linting `eslint`s internal folders, rather than `docblockr`s
I don't have access to a unix machine at the moment, so I can't really do further testing yet

File: lib/docsparser.js [hunk 1]
```diff
@@ -128,7 +128,7 @@ DocsParser.prototype.format_function = function(name, args, retval, options) {
     options = (typeof options !== 'undefined') ? options : {};
     var out = [];
     var i, len, format_str;
-    if(options.hasOwnProperty('as_setter')) {
+    if(Object.Object.prototype.hasOwnProperty.call(options, 'as_setter')) {
```
Comment (MoritzKn) [inline lib/docsparser.js]: You didn't mean `Object.Object`, did you?

**Comment by DevBonBon** [comment]
Sorry, should've probably started with that :D
I'm planning on adding the ability to customize the default docs generated and I'm trying to familiarize myself with the code. Updating dependencies and fixing everything that breaks afterwards is usually an easy place to start.
These also fix some vulnerabilities reported by npm.

**Comment by MoritzKn** [comment]
Alright updating the dependencies definitely sounds like a good idea. Though you should know that while docblockr isn't technically dead, nobody is actively working on it anymore. I am willing to merge PRs if they look good and I will also release a new version. But I currently don't have the time to put much work into this: exhaustive reviews, testing, etc.

Should you be interested in taking over the maintenance of this package, I think we could do that as well. But as you might already have noticed the code in this project got a little "rusty" and there is quite a bit of technical debt.
Should you still decide to put in the effort and maybe even clean up a little more, I am going to give my best to support your efforts.

I triggered a Travis build on master to see if the builds still parse in general. Please make sure everything is properly tested. I will trust you and won't test it again myself. If you are ready, I can merge this.

**Comment by DevBonBon** [comment]
I'm happy to "de-rust" the project, after using it for this long it's the least I can do, but I don't know about taking up maintaining it. I'll give it another thought after I'm done.

I'll just cleanup the commit history and request description and then I'm done.
I did some changes to how the tests are run, and removed the custom test runner script. I feel the `spec` reporter is too verbose for the Travis builds and didn't feel the script should be kept around just for the `*.spec.js` suffix. But if you don't feel that's appropriate, I can revert those changes.

**Comment by DevBonBon** [comment]
Ready to be merged now

**Comment by MoritzKn** [comment]
The maintenance thing was just an idea, no pressure. I am absolutely happy with you taking on the derusting.

Your work looks very clean :+1:

**Comment by MoritzKn** [comment]
I am not going to make a new release just yet, ok?

**Pull request merged**